src/overlays/actors/ovl_En_Firefly/z_en_firefly.c

```diff
@@ -21,64 +54,761 @@ const ActorInit En_Firefly_InitVars = {
+        this->invisible = true;
+    }
+
+    if (this->actor.params == KEESE_FIRE_FLY) {
+        this->auraType = KEESE_AURA_FIRE;
+        this->timer = Rand_S16Offset(20, 60);
+        this->actor.shape.rot.x = 0x1554;
+        this->actor.hintId = 0x11; // Fire Keese
+        this->maxAltitude = this->actor.home.pos.y;
+        this->actionFunc = EnFirefly_FlyIdle;
+    } else if (this->actor.params == KEESE_ICE_FLY) {
+        this->auraType = KEESE_AURA_ICE;
+        this->collider.info.toucher.effect = 2; // Freeze
+        this->actor.hintId = 0x56;              // Ice Keese
+        this->maxAltitude = this->actor.home.pos.y + 100.0f;
+        this->actionFunc = EnFirefly_FlyIdle;
+    } else {
+        this->auraType = KEESE_AURA_NONE;
+        this->collider.info.toucher.effect = 0; // Nothing
+        this->actor.hintId = 0x12;              // Keese
+        this->maxAltitude = this->actor.home.pos.y + 100.0f;
+        this->actionFunc = EnFirefly_Perch;
+    }
+
+    this->currentType = this->actor.params;
+    this->collider.dim.worldSphere.radius = sSphereInit.dim.modelSphere.radius;
+}
+
+void EnFirefly_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnFirefly* this = THIS;
+
+    Collider_DestroySphere(globalCtx, &this->collider);
+}
+
+void EnFirefly_SpawnIceEffects(EnFirefly* this, GlobalContext* globalCtx) {
+    if (this->unk_18F == 0xA) {
+        this->unk_18F = 0;
+        this->unk_2E8.x = 0.0f;
+        func_800BF7CC(globalCtx, &this->actor, &this->unk_2F8, 3, 2, 0.2f, 0.2f);
+    }
+}
+
+void EnFirefly_Extinguish(EnFirefly* this) {
+    this->currentType = KEESE_NORMAL;
+    this->collider.info.toucher.effect = 0; // Nothing
+    this->auraType = KEESE_AURA_NONE;
+    this->actor.hintId = 0x12; // Keese
+}
+
+void EnFirefly_Ignite(EnFirefly* this) {
+    if (this->actor.params == KEESE_FIRE_FLY) {
+        this->currentType = KEESE_FIRE;
+        this->collider.info.toucher.effect = 1; // Fire
+        this->auraType = KEESE_AURA_FIRE;
+        this->actor.hintId = 0x11; // Fire Keese
+    }
+}
+
+s32 EnFirefly_ReturnToPerch(EnFirefly* this, GlobalContext* globalCtx) {
+    ActorPlayer* player = PLAYER;
+    f32 distFromHome;
+
+    if (this->actor.params != KEESE_NORMAL_PERCH) {
+        return false;
+    }
+
+    if (Actor_XZDistanceToPoint(&player->base, &this->actor.home.pos) > 300.0f) {
+        distFromHome = Actor_DistanceToPoint(&this->actor, &this->actor.home.pos);
+
+        if (distFromHome < 5.0f) {
+            EnFirefly_SetupPerch(this);
+            return true;
+        }
+
+        if (distFromHome * 0.05f < 1.0f) {
+            this->actor.speedXZ *= distFromHome * 0.05f;
+        }
+
+        Math_ScaledStepToS(&this->actor.shape.rot.y, Actor_YawToPoint(&this->actor, &this->actor.home.pos), 0x300);
+        Math_ScaledStepToS(&this->actor.shape.rot.x, Actor_PitchToPoint(&this->actor, &this->actor.home.pos) + 0x1554,
+                           0x100);
```

Comment:
This wouldn't match with an `else if` `else` combo so you only have 1 `return true` would it?

Reply:
It does match, but I'd be inclined to keep the `return` after running the setup function. Be interested to hear what other people think.

Reply:
Personally I like just one return. I find it easier to see that if we enter `if (Actor_XZDistanceToPoint(&player->base, &this->actor.home.pos) > 300.0f)` then a true is returned.